gst/nnstreamer/tensor_typedef.h

```diff
@@ -219,4 +221,97 @@ typedef struct _GstTensorFilterProperties
+       */
+
+  int (*setInputDimension) (const GstTensorFilterProperties * prop,
+      void **private_data, const GstTensorsInfo * in_info,
+      GstTensorsInfo * out_info);
+      /**< Optional. Set Null if not supported. Tensor_filter::main will
+       * configure input dimension from pad-cap in run-time for the sub-plugin.
+       * Then, the sub-plugin is required to return corresponding output dimension
+       * If this is NULL, both getInput/OutputDimension must be non-NULL.
+       *
+       * When you use this, do NOT allocate or fix internal data structure based on it
+       * until invoke is called. Gstreamer may try different dimensions before
+       * settling down.
+       *
+       * @param[in] prop read-only property values
+       * @param[in/out] private_data A subplugin may save its internal private data here. The subplugin is responsible for alloc/free of this pointer.
+       * @param[in] in_info structure of input tensor info
+       * @param[out] out_info structure of output tensor info (return value)
+       * @return 0 if OK. non-zero if error.
+       */
+
+  int (*open) (const GstTensorFilterProperties * prop, void **private_data);
+      /**< Optional. tensor_filter.c will call this before any of other callbacks and will call once before calling close
+       *
+       * @param[in] prop read-only property values
+       * @param[in/out] private_data A subplugin may save its internal private data here. The subplugin is responsible for alloc/free of this pointer. Normally, open() allocates memory for private_data.
+       * @return 0 if ok. < 0 if error.
+       */
+
+  void (*close) (const GstTensorFilterProperties * prop, void **private_data);
+      /**< Optional. tensor_filter.c will not call other callbacks after calling close. Free-ing private_data is this function's responsibility. Set NULL after that.
+       *
+       * @param[in] prop read-only property values
+       * @param[in/out] private_data A subplugin may save its internal private data here. The subplugin is responsible for alloc/free of this pointer. Normally, close() frees private_data and set NULL.
+       */
+
+  void (*destroyNotify) (void * data);
+      /**< Optional. tensor_filter.c will call it when 'allocate_in_invoke' flag of the framework is TRUE. Basically, it is called when the data element is destroyed. If it's set as NULL, g_free() will be used as a default. It will be helpful when the data pointer is included as an object of a nnfw. For instance, if the data pointer is removed when the object is gone, it occurs error. In this case, the objects should be maintained for a while first and destroyed when the data pointer is destroyed. Those kinds of logic could be defined at this method.
+       *
+       * @param[in] data the data element.
+       */
+} GstTensorFilterFramework;
+
+/* extern functions for subplugin management, exist in tensor_filter.c */
+extern int tensor_filter_probe (GstTensorFilterFramework *tfsp);
```

Comment:
One question, I think these can be replaced with the functions - register_subplugin / unregister_subplugin in nnstreamer_subplugin.h.
Do you have a plan to remove functions (_probe/_exit) in filter and decoder?

Reply:
In the future, ```tensor_filter_probe`` should be modified to use ```register_sbplugin```.

Anyway, we need to keep ```tensor_filter_probe``` itself to support statically built subplugins.